src/sentry/monitors/serializers.py

```diff
@@ -330,14 +331,19 @@ def get_attrs(self, item_list, user, **kwargs):
         return attrs
 
     def serialize(self, obj, attrs, user, **kwargs) -> MonitorCheckInSerializerResponse:
+        config = obj.monitor_config.copy() if obj.monitor_config else {}
+        if "schedule_type" in config:
+            # XXX: We don't use monitor.get_schedule_type_display() in case it differs from the
+            # config saved on the check-in
+            config["schedule_type"] = ScheduleType.get_name(config["schedule_type"])
```

Comment:
Feels like what we probably really should be doing is having some type of sub-serializer for the config object, but this is fine for now I think